core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -258,7 +258,7 @@ public void run() {
   @Override
   public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(MethodDescriptor<ReqT, RespT> method,
       CallOptions callOptions) {
-    boolean hasCodecOverride = callOptions.getCompressor() == null;
+    boolean hasCodecOverride = callOptions.getCompressor() != null;
```

Comment:
You can see why we should add tests.  (me included!)  Its easy to add tests to an existing file, rather than start them up from scratch.